src/theory/bv/theory_bv_utils.cpp

```diff
@@ -26,21 +25,506 @@ namespace theory {
+         || kind == kind::BITVECTOR_XOR);
+  Assert(children.size() > 0);
+  if (children.size() == 1)
+  {
+    return children[0];
+  }
+  std::sort(children.begin(), children.end());
+  return NodeManager::currentNM()->mkNode(kind, children);
+}
+
+/* ------------------------------------------------------------------------- */
+
+Node mkNot(Node child)
+{
+  return NodeManager::currentNM()->mkNode(kind::NOT, child);
+}
+
+Node mkAnd(TNode node1, TNode node2)
+{
+  return NodeManager::currentNM()->mkNode(kind::AND, node1, node2);
+}
+
+Node mkAnd(const std::vector<TNode>& conjunctions)
+{
+  std::set<TNode> all;
+  all.insert(conjunctions.begin(), conjunctions.end());
+
+  if (all.size() == 0)
+  {
+    return mkTrue();
+  }
+
+  if (all.size() == 1)
+  {
+    // All the same, or just one
+    return conjunctions[0];
+  }
+
+  NodeBuilder<> conjunction(kind::AND);
+  std::set<TNode>::const_iterator it = all.begin();
+  std::set<TNode>::const_iterator it_end = all.end();
+  while (it != it_end)
+  {
+    conjunction << *it;
+    ++it;
+  }
+
+  return conjunction;
+}
+
+Node mkAnd(const std::vector<Node>& conjunctions)
+{
+  std::set<TNode> all;
+  all.insert(conjunctions.begin(), conjunctions.end());
+
+  if (all.size() == 0)
+  {
+    return mkTrue();
+  }
+
+  if (all.size() == 1)
+  {
+    // All the same, or just one
+    return conjunctions[0];
+  }
+
+  NodeBuilder<> conjunction(kind::AND);
+  std::set<TNode>::const_iterator it = all.begin();
+  std::set<TNode>::const_iterator it_end = all.end();
+  while (it != it_end)
+  {
+    conjunction << *it;
+    ++it;
+  }
+
+  return conjunction;
+}
+
+Node mkOr(TNode node1, TNode node2)
+{
+  return NodeManager::currentNM()->mkNode(kind::OR, node1, node2);
+}
+
+Node mkOr(const std::vector<Node>& nodes)
+{
+  std::set<TNode> all;
+  all.insert(nodes.begin(), nodes.end());
+
+  if (all.size() == 0)
+  {
+    return mkTrue();
+  }
+
+  if (all.size() == 1)
+  {
+    // All the same, or just one
+    return nodes[0];
+  }
+
+  NodeBuilder<> disjunction(kind::OR);
+  std::set<TNode>::const_iterator it = all.begin();
+  std::set<TNode>::const_iterator it_end = all.end();
+  while (it != it_end)
+  {
+    disjunction << *it;
+    ++it;
+  }
+
+  return disjunction;
+}
+
+Node mkXor(TNode node1, TNode node2)
+{
+  return NodeManager::currentNM()->mkNode(kind::XOR, node1, node2);
+}
+
+/* ------------------------------------------------------------------------- */
+
+Node mkSignExtend(TNode node, unsigned amount)
+{
+  NodeManager* nm = NodeManager::currentNM();
+  Node signExtendOp =
+      nm->mkConst<BitVectorSignExtend>(BitVectorSignExtend(amount));
+  return nm->mkNode(signExtendOp, node);
+}
+
+/* ------------------------------------------------------------------------- */
+
+Node mkExtract(TNode node, unsigned high, unsigned low)
+{
+  Node extractOp = NodeManager::currentNM()->mkConst<BitVectorExtract>(
+      BitVectorExtract(high, low));
+  std::vector<Node> children;
```

Comment:
Nit: no need to use a vector here, just call `NodeManager::currentNM()->mkNode(extractOp, node)`